src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/FlagAuthoritySystem.java

```diff
@@ -113,4 +120,38 @@ public void onGiveItemToCharacterHoldItem(GiveItemEvent event, EntityRef item, I
                 delayManager.cancelDelayedAction(item, LASUtils.DROPPED_FLAG);
         }
     }
+
+    @ReceiveEvent
+    public void dropFlagRequest(FlagDropRequestEvent event, EntityRef targetPlayer) {
```

Comment:
naming nitpick: why "request" here and only here? theoretically, "move flag to base" and "give flag to player" also are mere requests. I'd say we drop the "Request" suffix here to keep it consistent.

Reply:
The main reason I did that over here is because of the overlap with the FlagDrop event. The FlagDrop event is triggered when the flag drops and to avoid any conflicts I added the Request suffix to indicate that the flag should be dropped when this event is sent.

Reply:
The "request" to drop a flag would be a trigger event while the "flag drops now / was just dropped" is more of a notification event. Let's refer to the documentation linked above and adjust the naming and event implementation accordingly, I'd say.